src/engine/Bind.cpp

```diff
@@ -149,6 +149,8 @@ Result Bind::computeResult(bool requestLaziness) {
       [](auto applyBind,
          std::shared_ptr<const Result> result) -> Result::Generator {
     for (auto& [idTable, localVocab] : result->idTables()) {
+      // Introduce a fresh primary word set
```

Comment:
More "The `LocalVocab` disallows inserts if it doesn't own its `primaryWordSet` exclusively. We clone the local vocab to enforce this invariant in all cases".